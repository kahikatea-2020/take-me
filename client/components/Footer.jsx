import React from 'react'
import { Menu, Container } from 'semantic-ui-react'
import { Link } from 'react-router-dom'

class Footer extends React.Component {
  render () {
    return (
      <>
        <Container id="footer">
          <div className="ui grid container" id="footerWrapper">
            <div className="four wide column" id="footerColumn">
              <div className="ui list" id="listWrapper">
                <Link to={'/'}>
                  <div className="item">
                    <img src="/footer_blue-15.png" alt="logo" id="footerLogo"/>
                  </div>
                </Link>
                <div className="item">
                  <p id="tagLine">Doing our part for a more sustainable <br />New Zealand.</p>
                </div>
              </div>
            </div>
            <div className="four wide column">
              <div className="ui link list" id="footerLinks">
                <div className="active item" id="footerHeaders">
                  <Link className="item" id="footerHeaders" to={`/`}>Home</Link>
                </div>
                <Link className="item" id="footerLinks" to={`/listings/14`}>About</Link>
                <Link className="item" id="footerLinks" to={`/listings/14`}>Guidelines</Link>
                <Link className="item" id="footerLinks" to={`/sign-up`}>Sign Up</Link>
              </div>
            </div>
            <div className="four wide column">
              <div className="ui link list" id="footerLinks">
                <div className="active item" id="footerHeaders">
                  <Link className="item" id="footerHeaders" to={`/`}>Categories</Link>
                </div>
                <Link className="item" id="footerLinks" to={`/listings/14`}>Latest Listings</Link>
                <Link className="item" id="footerLinks" to={`/listings/14`}>By Location</Link>
                <Link className="item" id="footerLinks" to={`/new-listing`}>List an Item</Link>
              </div>
            </div>
            <div className="four wide column">
              <div className="ui list" id="listWrapper">
                <div className="item">
                  <i className="marker icon"></i>
                  <div className="content">
                    New Zealand
                  </div>
                </div>
                <div className="item">
                  <i className="mail icon"></i>
                  <div className="content">
                    <a href="mailto:hello@takemenz.com">hello@takemenz.com</a>
                  </div>
                </div>
                <div className="item">
                  <i className="linkify icon"></i>
                  <div className="content">
                    <a href="https://take-me-nz.herokuapp.com">takemenz.com</a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Container>
      </>
    )
  }
}

export default Footer
