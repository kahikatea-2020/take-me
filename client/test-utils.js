import React from 'react'
import { Provider } from 'react-redux'
import { createStore, applyMiddleware, compose } from 'redux'
import { MemoryRouter as Router } from 'react-router-dom'
import thunkMiddleware from 'redux-thunk'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import reducer from './reducers'
const enhancer = compose(applyMiddleware(thunkMiddleware))

export const renderWithRedux = (
  ui,
  { initialState, initialEntries = ['/'], store = createStore(reducer, initialState, enhancer) } = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router initialEntries={initialEntries} initialIndex={0}>
          {ui}
        </Router>
      </Provider>
    ),
    store
  }
}
