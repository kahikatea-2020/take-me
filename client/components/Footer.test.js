import React from 'react'
import { renderWithRedux } from '../test-utils'
import '@testing-library/jest-dom/extend-expect'

import Footer from './Footer'

test('Test Footer renders correctly', () => {
  const rendered = renderWithRedux(<Footer />)
  expect(rendered).toMatchSnapshot()
})