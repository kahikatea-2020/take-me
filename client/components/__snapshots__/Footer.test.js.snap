// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Footer renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui container"
        id="footer"
      >
        <div
          class="ui grid container"
          id="footerWrapper"
        >
          <div
            class="four wide column"
            id="footerColumn"
          >
            <div
              class="ui list"
              id="listWrapper"
            >
              <a
                href="/"
              >
                <div
                  class="item"
                >
                  <img
                    alt="logo"
                    id="footerLogo"
                    src="takeme-green.png"
                  />
                </div>
              </a>
              <div
                class="item"
              >
                <p
                  id="tagLine"
                >
                  Doing our part for a more sustainable 
                  <br />
                  New Zealand.
                </p>
              </div>
            </div>
          </div>
          <div
            class="four wide column"
          >
            <div
              class="ui link list"
              id="footerLinks"
            >
              <div
                class="active item"
                id="footerHeaders"
              >
                <a
                  class="item"
                  href="/"
                  id="footerHeaders"
                >
                  Home
                </a>
              </div>
              <a
                class="item"
                href="/listings/14"
                id="footerLinks"
              >
                About
              </a>
              <a
                class="item"
                href="/listings/14"
                id="footerLinks"
              >
                Guidelines
              </a>
              <a
                class="item"
                href="/sign-up"
                id="footerLinks"
              >
                Sign Up
              </a>
            </div>
          </div>
          <div
            class="four wide column"
          >
            <div
              class="ui link list"
              id="footerLinks"
            >
              <div
                class="active item"
                id="footerHeaders"
              >
                <a
                  class="item"
                  href="/"
                  id="footerHeaders"
                >
                  Categories
                </a>
              </div>
              <a
                class="item"
                href="/listings/14"
                id="footerLinks"
              >
                Latest Listings
              </a>
              <a
                class="item"
                href="/listings/14"
                id="footerLinks"
              >
                By Location
              </a>
              <a
                class="item"
                href="/new-listing"
                id="footerLinks"
              >
                List an Item
              </a>
            </div>
          </div>
          <div
            class="four wide column"
          >
            <div
              class="ui list"
              id="listWrapper"
            >
              <div
                class="item"
              >
                <i
                  class="marker icon"
                />
                <div
                  class="content"
                >
                  New Zealand
                </div>
              </div>
              <div
                class="item"
              >
                <i
                  class="mail icon"
                />
                <div
                  class="content"
                >
                  <a
                    href="mailto:hello@takemenz.com"
                  >
                    hello@takemenz.com
                  </a>
                </div>
              </div>
              <div
                class="item"
              >
                <i
                  class="linkify icon"
                />
                <div
                  class="content"
                >
                  <a
                    href="https://take-me-nz.herokuapp.com"
                  >
                    takemenz.com
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui container"
      id="footer"
    >
      <div
        class="ui grid container"
        id="footerWrapper"
      >
        <div
          class="four wide column"
          id="footerColumn"
        >
          <div
            class="ui list"
            id="listWrapper"
          >
            <a
              href="/"
            >
              <div
                class="item"
              >
                <img
                  alt="logo"
                  id="footerLogo"
                  src="takeme-green.png"
                />
              </div>
            </a>
            <div
              class="item"
            >
              <p
                id="tagLine"
              >
                Doing our part for a more sustainable 
                <br />
                New Zealand.
              </p>
            </div>
          </div>
        </div>
        <div
          class="four wide column"
        >
          <div
            class="ui link list"
            id="footerLinks"
          >
            <div
              class="active item"
              id="footerHeaders"
            >
              <a
                class="item"
                href="/"
                id="footerHeaders"
              >
                Home
              </a>
            </div>
            <a
              class="item"
              href="/listings/14"
              id="footerLinks"
            >
              About
            </a>
            <a
              class="item"
              href="/listings/14"
              id="footerLinks"
            >
              Guidelines
            </a>
            <a
              class="item"
              href="/sign-up"
              id="footerLinks"
            >
              Sign Up
            </a>
          </div>
        </div>
        <div
          class="four wide column"
        >
          <div
            class="ui link list"
            id="footerLinks"
          >
            <div
              class="active item"
              id="footerHeaders"
            >
              <a
                class="item"
                href="/"
                id="footerHeaders"
              >
                Categories
              </a>
            </div>
            <a
              class="item"
              href="/listings/14"
              id="footerLinks"
            >
              Latest Listings
            </a>
            <a
              class="item"
              href="/listings/14"
              id="footerLinks"
            >
              By Location
            </a>
            <a
              class="item"
              href="/new-listing"
              id="footerLinks"
            >
              List an Item
            </a>
          </div>
        </div>
        <div
          class="four wide column"
        >
          <div
            class="ui list"
            id="listWrapper"
          >
            <div
              class="item"
            >
              <i
                class="marker icon"
              />
              <div
                class="content"
              >
                New Zealand
              </div>
            </div>
            <div
              class="item"
            >
              <i
                class="mail icon"
              />
              <div
                class="content"
              >
                <a
                  href="mailto:hello@takemenz.com"
                >
                  hello@takemenz.com
                </a>
              </div>
            </div>
            <div
              class="item"
            >
              <i
                class="linkify icon"
              />
              <div
                class="content"
              >
                <a
                  href="https://take-me-nz.herokuapp.com"
                >
                  takemenz.com
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;

exports[`Test New Listing component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui container"
        id="footer"
      >
        <div
          class="ui grid container"
          id="footerWrapper"
        >
          <div
            class="four wide column"
            id="footerColumn"
          >
            <div
              class="ui list"
              id="listWrapper"
            >
              <a
                href="/"
              >
                <div
                  class="item"
                >
                  <img
                    alt="logo"
                    id="footerLogo"
                    src="takeme-green.png"
                  />
                </div>
              </a>
              <div
                class="item"
              >
                <p
                  id="tagLine"
                >
                  Doing our part for a more sustainable 
                  <br />
                  New Zealand.
                </p>
              </div>
            </div>
          </div>
          <div
            class="four wide column"
          >
            <div
              class="ui link list"
              id="footerLinks"
            >
              <div
                class="active item"
                id="footerHeaders"
              >
                <a
                  class="item"
                  href="/"
                  id="footerHeaders"
                >
                  Home
                </a>
              </div>
              <a
                class="item"
                href="/listings/14"
                id="footerLinks"
              >
                About
              </a>
              <a
                class="item"
                href="/listings/14"
                id="footerLinks"
              >
                Guidelines
              </a>
              <a
                class="item"
                href="/sign-up"
                id="footerLinks"
              >
                Sign Up
              </a>
            </div>
          </div>
          <div
            class="four wide column"
          >
            <div
              class="ui link list"
              id="footerLinks"
            >
              <div
                class="active item"
                id="footerHeaders"
              >
                <a
                  class="item"
                  href="/"
                  id="footerHeaders"
                >
                  Categories
                </a>
              </div>
              <a
                class="item"
                href="/listings/14"
                id="footerLinks"
              >
                Latest Listings
              </a>
              <a
                class="item"
                href="/listings/14"
                id="footerLinks"
              >
                By Location
              </a>
              <a
                class="item"
                href="/new-listing"
                id="footerLinks"
              >
                List an Item
              </a>
            </div>
          </div>
          <div
            class="four wide column"
          >
            <div
              class="ui list"
              id="listWrapper"
            >
              <div
                class="item"
              >
                <i
                  class="marker icon"
                />
                <div
                  class="content"
                >
                  New Zealand
                </div>
              </div>
              <div
                class="item"
              >
                <i
                  class="mail icon"
                />
                <div
                  class="content"
                >
                  <a
                    href="mailto:hello@takemenz.com"
                  >
                    hello@takemenz.com
                  </a>
                </div>
              </div>
              <div
                class="item"
              >
                <i
                  class="linkify icon"
                />
                <div
                  class="content"
                >
                  <a
                    href="https://take-me-nz.herokuapp.com"
                  >
                    takemenz.com
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui container"
      id="footer"
    >
      <div
        class="ui grid container"
        id="footerWrapper"
      >
        <div
          class="four wide column"
          id="footerColumn"
        >
          <div
            class="ui list"
            id="listWrapper"
          >
            <a
              href="/"
            >
              <div
                class="item"
              >
                <img
                  alt="logo"
                  id="footerLogo"
                  src="takeme-green.png"
                />
              </div>
            </a>
            <div
              class="item"
            >
              <p
                id="tagLine"
              >
                Doing our part for a more sustainable 
                <br />
                New Zealand.
              </p>
            </div>
          </div>
        </div>
        <div
          class="four wide column"
        >
          <div
            class="ui link list"
            id="footerLinks"
          >
            <div
              class="active item"
              id="footerHeaders"
            >
              <a
                class="item"
                href="/"
                id="footerHeaders"
              >
                Home
              </a>
            </div>
            <a
              class="item"
              href="/listings/14"
              id="footerLinks"
            >
              About
            </a>
            <a
              class="item"
              href="/listings/14"
              id="footerLinks"
            >
              Guidelines
            </a>
            <a
              class="item"
              href="/sign-up"
              id="footerLinks"
            >
              Sign Up
            </a>
          </div>
        </div>
        <div
          class="four wide column"
        >
          <div
            class="ui link list"
            id="footerLinks"
          >
            <div
              class="active item"
              id="footerHeaders"
            >
              <a
                class="item"
                href="/"
                id="footerHeaders"
              >
                Categories
              </a>
            </div>
            <a
              class="item"
              href="/listings/14"
              id="footerLinks"
            >
              Latest Listings
            </a>
            <a
              class="item"
              href="/listings/14"
              id="footerLinks"
            >
              By Location
            </a>
            <a
              class="item"
              href="/new-listing"
              id="footerLinks"
            >
              List an Item
            </a>
          </div>
        </div>
        <div
          class="four wide column"
        >
          <div
            class="ui list"
            id="listWrapper"
          >
            <div
              class="item"
            >
              <i
                class="marker icon"
              />
              <div
                class="content"
              >
                New Zealand
              </div>
            </div>
            <div
              class="item"
            >
              <i
                class="mail icon"
              />
              <div
                class="content"
              >
                <a
                  href="mailto:hello@takemenz.com"
                >
                  hello@takemenz.com
                </a>
              </div>
            </div>
            <div
              class="item"
            >
              <i
                class="linkify icon"
              />
              <div
                class="content"
              >
                <a
                  href="https://take-me-nz.herokuapp.com"
                >
                  takemenz.com
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
